### PR TITLE
feat(api): add merge request pipeline manager and deprecate mr.pipelines() method

### PR DESCRIPTION
**Pull request by nejch**
Closes https://github.com/python-gitlab/python-gitlab/issues/1312.
Closes https://github.com/python-gitlab/python-gitlab/issues/1239.

~~todo: technically this breaks the CLI `gitlab project-merge-request pipelines` and requires `project-merge-request-pipelines list` instead. I'll see if I can hack the `register_custom_action` decorator to accept custom action names, not just the current method name.~~ I had to remove it as there would be a namespace clash between the old method and the new manager.

Well, I've managed to hack `register_custom_action` a little so it takes an optional `custom_action` argument to override the current function name as the CLI action, so should be backwards compatible, but then realized the old method probably never worked in CLI due to missing CLI args :facepalm: 

Still, now you can do (until 3.0.0 or so, but probably useless anyway):

```shell
gitlab project-merge-request pipelines --help
usage: gitlab project-merge-request pipelines [-h] --project-id PROJECT_ID [--sudo SUDO] --iid IID

optional arguments:
  -h, --help            show this help message and exit
  --project-id PROJECT_ID
  --sudo SUDO
  --iid IID
```

And the new list, with proper help and args:
```shell
gitlab project-merge-request-pipeline list --help
usage: gitlab merge-request-pipeline list [-h] [--sudo SUDO] --project-id PROJECT_ID --mr-iid MR_IID
                                          [--page PAGE] [--per-page PER_PAGE] [--all]

optional arguments:
  -h, --help            show this help message and exit
  --sudo SUDO
  --project-id PROJECT_ID
  --mr-iid MR_IID
  --page PAGE
  --per-page PER_PAGE
  --all
```

### CODE REVIEW

**Comment by nejch** [comment]
Maybe @JohnVillalovos  you want to have a look at the typing here as well, I always forget decorators and have to google the wrap functions every time I work with them :rofl:

**Comment by JohnVillalovos** [comment]
> 
> 
> Maybe @JohnVillalovos you want to have a look at the typing here as well, I always forget decorators and have to google the wrap functions every time I work with them 🤣

From my initial look it seems correct. Also mypy is checking that `gitlab/cli.py` file, so seems good.

Looks Good To Me (LGTM) 👍

Thanks!

**Comment by max-wittig** [comment]
We're most likely gonna go to 3.0.0 directly, because of this breaking change: https://github.com/python-gitlab/python-gitlab/pull/1278. Maybe have those two in one release would be nice 😄

**Comment by msune** [comment]
:+1: on this.

I seem to be hitting a limitation/bug of `pipelines()` https://github.com/python-gitlab/python-gitlab/issues/1312#issuecomment-808813678 and in another issue that I will not bother mentioning.

Any chance this could be integrated soon?

**Comment by max-wittig** [comment]
@nejch Sorry for making you wait for long that everything conflicts. I just see like 15 notifications everyday from this project. You people are crazy 😄 

Would you mind to rebase?

File: gitlab/cli.py
Comment (JohnVillalovos) [inline gitlab/cli.py:58]: I don't think this function definition needs to be updated.

`__F` already declares it a callable that takes any number of arguments.

**Review by JohnVillalovos — CHANGES_REQUESTED** [review]
Overall looks good to me. Left one thing I think should be fixed.

**Review by JohnVillalovos — APPROVED** [review]
Thanks. This looks good.

I'll let you merge it or if you want to wait for @max-wittig to review. Your choice.

File: gitlab/cli.py
Comment (nejch) [reply gitlab/cli.py]: Oops, thanks for this :)

**Pull request merged**
